src/Microsoft.TemplateEngine.Abstractions/IEngineEnvironmentSettings.cs

```diff
@@ -35,5 +35,10 @@ public interface IEngineEnvironmentSettings : IDisposable
         /// Component manager for this instance of settings.
         /// </summary>
         IComponentManager Components { get; }
+
+        /// <summary>
+        /// Flag specifies if tab completion mode is enabled.
+        /// </summary>
+        bool TabCompletionMode { get; }
```

Comment:
This looks as too specific for `IEngineEnvironmentSettings`, also causes breaking changes.
Please consider moving to `TemplatePackageManager` instead or maybe to `IEnvironment`.

Reply:
done